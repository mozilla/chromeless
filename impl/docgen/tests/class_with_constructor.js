/**
 * A test of basic class parsing
 */

/**
 * @class URL
 * A class which parses a url and exposes its various
 * components separately.
 */

/**
 * @constructor
 * This is the constructor of the class
 * @param {foo} bar it accepts a bar argument of type foo
 */

/** @endclass */
