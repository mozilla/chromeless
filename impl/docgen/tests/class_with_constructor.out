{
  "classes": [
    {
      "constructor": {
        "desc": "This is the constructor of the class", 
        "params": [
          {
            "desc": "it accepts a bar argument of type foo", 
            "name": "bar", 
            "type": "foo"
          }
        ]
      }, 
      "desc": "A class which parses a url and exposes its various\ncomponents separately.", 
      "name": "URL"
    }
  ], 
  "desc": "A test of basic class parsing", 
  "filename": "tests/class_with_constructor.js", 
  "module": "class_with_constructor"
}
